infrahub_sdk/client.py

```diff
@@ -893,13 +894,20 @@ async def execute_graphql(
             branch_name (str, optional): Name of the branch on which the query will be executed. Defaults to None.
             at (str, optional): Time when the query should be executed. Defaults to None.
             timeout (int, optional): Timeout in second for the query. Defaults to None.
-            raise_for_error (bool, optional): Flag to indicate that we need to raise an exception if the response has some errors. Defaults to True.
+            raise_for_error (bool, optional): Deprecated, flag to indicate that we need to raise an exception if the response has some errors.
+            Defaults to True.
         Raises:
             GraphQLError: _description_
 
         Returns:
             _type_: _description_
         """
+        if not raise_for_error:
+            warnings.warn(
+                "Setting `raise_for_error=False` is deprecated, use `try/except` to handle errors.",
+                DeprecationWarning,
+                stacklevel=1,
+            )
```

Comment:
I'm wondering if we should instead change the option to be `raise_for_error: bool | None` and then have:

```python
        if raise_for_error is not None:
            warnings.warn(
                "Setting `raise_for_error` is deprecated, use `try/except` to handle errors.",
                DeprecationWarning,
                stacklevel=1,
            )
```

I.e. we also want to deprecate setting raise_for_error=True. Does that make sense?